src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,495 @@
+                await reader.ReadToEndAsync();
+            }
+            catch (Exception e)
+            {
+                Log($"Failed to send the response signature to {url}, as {e.Message}", LogLevel.Warning);
+            }
+        }
+
+        private async Task SendResponseSignatureAsync(ulong requestId, byte[] txSign, KeyPair keyPair)
+        {
+            var message = keyPair.PublicKey.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+            var sign = Crypto.Sign(message, keyPair.PrivateKey, keyPair.PublicKey.EncodePoint(false)[1..]);
+            var param = "\"" + Convert.ToBase64String(keyPair.PublicKey.ToArray()) + "\", " + requestId + ", \"" + Convert.ToBase64String(txSign) + "\",\"" + Convert.ToBase64String(sign) + "\"";
+            var content = "{\"id\":" + Interlocked.Increment(ref counter) + ",\"jsonrpc\":\"2.0\",\"method\":\"submitoracleresponse\",\"params\":[" + param + "]}";
+
+            var tasks = Settings.Default.Nodes.Select(p => SendContentAsync(p, content));
+            await Task.WhenAll(tasks);
+        }
+
+        private async Task ProcessRequestAsync(StoreView snapshot, OracleRequest req)
+        {
+            Log($"Process oracle request: {req}, txid: {req.OriginalTxid}, url: {req.Url}");
+
+            (OracleResponseCode code, string data) = await ProcessUrlAsync(req.Url);
+
+            var oracleNodes = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.Oracle, snapshot.Height + 1);
+            foreach (var (requestId, request) in NativeContract.Oracle.GetRequestsByUrl(snapshot, req.Url))
+            {
+                var result = Array.Empty<byte>();
+                if (code == OracleResponseCode.Success)
+                {
+                    try
+                    {
+                        result = Filter(data, request.Filter);
+                    }
+                    catch
+                    {
+                        code = OracleResponseCode.Error;
+                    }
+                }
+                var response = new OracleResponse() { Id = requestId, Code = code, Result = result };
+                var responseTx = CreateResponseTx(snapshot, request, response, oracleNodes);
+                var backupTx = CreateResponseTx(snapshot, request, new OracleResponse() { Code = OracleResponseCode.ConsensusUnreachable, Id = requestId, Result = Array.Empty<byte>() }, oracleNodes);
+
+                Log($"Builded response tx:{responseTx.Hash} requestTx:{request.OriginalTxid} requestId: {requestId}");
+
+                List<Task> tasks = new List<Task>();
+                ECPoint[] oraclePublicKeys = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.Oracle, snapshot.Height + 1);
+                foreach (var account in wallet.GetAccounts())
+                {
+                    var oraclePub = account.GetKey().PublicKey;
```

Comment:
should check account.GetKey() is null. otherwise it will throw exceptions.

Reply:
Fixed